File: packages_mjs/runtime-template-resolver/src/patterns.ts
export const PATTERNS = {
    // $.path.to.value
    DOT_PATH: /\$\.([a-zA-Z_][\w]*(?:\.[\w]+|\[\d+\]|\["[^"]+"\]|\[\'[^ \']+\'\])*)/g,

    // {{path.to.value}} or {{path|"default"}}
    MUSTACHE: /\{\{([^}|]+)(?:\|"([^"]*)")?\}\}/g,

    // Escaped variants
    ESCAPED_DOT: /\\\$\./g,
    ESCAPED_MUSTACHE: /\\\{\{/g,
};

File: packages_py/runtime_template_resolver/src/runtime_template_resolver/patterns.py
import re

PATTERNS = {
    # $.path.to.value
    "DOT_PATH": re.compile(r'\$\.([a-zA-Z_]\w*(?:\.\w+|\[\d+\]|\["[^"]+"\]|\[\'[^ \']+\'\])*)'),

    # {{path.to.value}} or {{path|"default"}}
    "MUSTACHE": re.compile(r'\{\{([^}|]+)(?:\|"([^"]*)")?\}\}'),

    # Escaped variants
    "ESCAPED_DOT": re.compile(r'\\\$\.'),
    "ESCAPED_MUSTACHE": re.compile(r'\\\{\{'),
}
